glommio/src/sync/rwlock.rs

```diff
@@ -763,17 +757,23 @@ impl<T> RwLock<T> {
     ///     dead_locker.await;
     /// });
     /// ```
-    pub fn close(&self) {
+    pub fn close(&self) -> LockResult<()> {
         let mut state = self.state.borrow_mut();
         if state.closed {
-            return;
+            return Ok(());
+        }
+
+        if state.writers > 0 || state.readers > 0 {
+            return Err(GlommioError::CanNotBeClosed(
+                ResourceType::RwLock,
+                "Lock is still hold by fibers",
```

Comment:
*"held" and also maybe have it say "fiber(s)" so that it's clear it also might be a single one